### ISSUE TITLE
mvkWaitForFences returns VK_SUCCESS with timeout 0

### ISSUE THREAD

**Issue by freesig** [KhronosGroup/MoltenVK#429]
`mvkWaitForFences` will always return `VK_SUCCESS` with a timeout of 0 even if the fence isn't ready.
```c++
VkResult mvkWaitForFences(uint32_t fenceCount,
						  const VkFence* pFences,
						  VkBool32 waitAll,
						  uint64_t timeout) {

	VkResult rslt = VK_SUCCESS;
	MVKFenceSitter fenceSitter(waitAll);

	for (uint32_t i = 0; i < fenceCount; i++) {
		((MVKFence*)pFences[i])->addSitter(&fenceSitter);
	}

	if ( !fenceSitter.wait(timeout) && timeout > 0 ) {
		rslt = mvkNotifyErrorWithText(VK_TIMEOUT, "Vulkan fence timeout after %llu nanoseconds.", timeout);
	}

	for (uint32_t i = 0; i < fenceCount; i++) {
		((MVKFence*)pFences[i])->removeSitter(&fenceSitter);
	}

	return rslt;
}
```
`&& timeout > 0 ` means that rslt can never be set to VK_TIMEOUT when timeout is 0.
The vulkan spec states:

> If timeout is zero, then vkWaitForFences does not wait, but simply returns the current state of the fences. VK_TIMEOUT will be returned in this case if the condition is not satisfied, even though no actual wait was performed.

**Comment by billhollings** [comment]
Thanks for pointing this out.

This resulted from a typo during recent streamlining of fence handling.

Fixed now in PR #433.

`VK_TIMEOUT` is returned even on zero wait. And the error is only logged if the timeout is greater than zero.

Please retest with latest code and close this issue if fixed.

**Comment by freesig** [comment]
yep this works now
